presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -672,7 +690,7 @@ private Function<Object, Object> getColumnIndexConversions(PrimitiveType type)
                 case BOOLEAN:
                     return buffer -> ((ByteBuffer) buffer).get(0) != 0;
                 case INT32:
-                    return buffer -> ((ByteBuffer) buffer).order(LITTLE_ENDIAN).getInt(0);
+                    return buffer -> (long) ((ByteBuffer) buffer).order(LITTLE_ENDIAN).getInt(0);
```

Comment:
Can you help me understand why this is needed?

Reply:
![image](https://github.com/prestodb/presto/assets/31331967/0c99a15a-a733-4a62-9a98-f952cd8b22d3)
Because the `getValuesAsLong` method requires a Long object. The code before the modification will return an Integer object, which will cause an error.